components/fatfs/src/vfs_fat.c

```diff
@@ -307,6 +307,16 @@ static int vfs_fat_open(void* ctx, const char * path, int flags, int mode)
         goto out;
     }
 out:
+    if ((fd >= 0) && (flags & O_APPEND)) {
```

Comment:
The `out:` block is intended for cleanup (common C pattern), so it reads better if this `if (flags & O_APPEND)` block sits above it. At that point fd will always be valid so you can remove the `>= 0` test (as the function skips to `out` if there's been an error).

Reply:
Of course, done.